crates/rune/src/diagnostics/emit.rs

```diff
@@ -475,6 +535,63 @@ where
     Ok(())
 }
 
+/// Helper to emit diagnostics for a runtime warning.
+fn runtime_warning_diagnostics_emit<O>(
+    this: &RuntimeWarningDiagnostic,
+    out: &mut O,
+    sources: &Sources,
+    config: &term::Config,
+    debug_info: Option<&DebugInfo>,
+    context: Option<&Context>,
+) -> Result<(), EmitError>
+where
+    O: WriteColor,
+{
+    let mut notes = ::rust_alloc::vec::Vec::new();
+    let mut labels = ::rust_alloc::vec::Vec::new();
+    let mut message = String::new();
+
+    match this.kind {
+        RuntimeWarningDiagnosticKind::UsedDeprecated { hash } => {
+            // try to get the function name - this needs to be improved
```

Comment:
This works most of the time, but not for instance fields. `DebugInfo` only holds the information from source files and not static generated infos. This is okay for me at this time but maybe you have a nice idea.

Reply:
Eh, that's tough. We'd have to emit a different deprecated method which includes the type hash so that we can reverse the hash, like [we do here](https://github.com/rune-rs/rune/blob/25b256ee06a6d3cbd5e45f647c7d809f7fb3f8f6/crates/rune/src/diagnostics/emit.rs#L206).

It would require a bit more work, and would be an OK future change.